rosbag2_transport/include/rosbag2_transport/play_options.hpp

```diff
@@ -40,6 +40,10 @@ struct PlayOptions
   std::unordered_map<std::string, rclcpp::QoS> topic_qos_profile_overrides = {};
   bool loop = false;
   std::vector<std::string> topic_remapping_options = {};
+
+  // Rate in Hz at which to publish to /clock.
+  // 0 (or negative) means that no publisher will be created
+  double clock_publish_frequency = 0.0;
```

Comment:
This wouldn't be appropriate for `PlayOptions` - because the default should be "do not publish /clock". This consideration should be at the CLI argument parsing level. Maybe something like:

```
parser.add_argument('--clock', nargs='?', const=40, default=0)
```

With the above, then just `--clock` would mean 40Hz. But if it is not provided, then 0Hz. Otherwise, if an argument is provided, then use that.

Reply:
I've update the CLI to work this way

Reply:
Thanks.